src/ros/neem_validation.pl

```diff
@@ -0,0 +1,155 @@
+
+
+get_actions_without_participants(ActionsWithoutParticipants) :-
+  findall(Action, 
+    ( is_action(Action), 
+    \+ has_participant(Action, P)), 
+    ActionsWithoutParticipants),
+  length(ActionsWithoutParticipants, Listlength),
+  ( Listlength > 0 -> print_message(warning, 'The following actions have no participants. Please assert the participants involved in the action'),
+  forall(member(Action, ActionsWithoutParticipants), print_message(warning, Action))), !;
+  print_message(info, 'All actions have participants'), true.
+
+
+get_child_frames_without_link_to_world_frame(World, ChildFramesWithoutLinkToWorld) :-
+  mng_db_name(DB),
+  mng_distinct_values(DB, tf, 'child_frame_id', ChildFrames),
+  findall(Frame, 
+    (member(Frame, ChildFrames), 
+    \+ is_at(Frame, [World, Translation, Rotation])),
+    ChildFramesWithoutLinkToWorld),
+  length(ChildFramesWithoutLinkToWorld, Listlength),
+  ( Listlength > 0 -> print_message(warning, 'The following child frames have no link to the world frame'),
+  forall(member(Child, ChildFramesWithoutLinkToWorld), print_message(warning, Child))), !;
+  print_message(info, 'All the child frames in the TF are fine'), true.
+
+get_objects_without_location(POWithoutLocation) :-
+   findall(Object,
+    ( is_physical_object(Object),
+    \+ object_localization(Object, Location)),
+    POWithoutLocation),
+  length(POWithoutLocation, Listlength),
+  ( Listlength > 0 -> print_message(warning, 'The following physical objects have no location. Please assert the locations for these objects'),
+  forall(member(O, POWithoutLocation), print_message(warning, O))), !;
+   print_message(info, 'All objects have a location'), true.
+
+get_objects_without_shape(POWithoutShape) :-
+   findall(Object,
+    ( is_physical_object(Object),
+    (\+ object_shape(Object, Shape, Pos, Material); \+ ros_urdf:object_shape(Object, Shape, Pos, Material))),
+    PO),
+  list_to_set(PO, POWithoutShape),
+  length(POWithoutShape, Listlength),
+  ( Listlength > 0 -> print_message(warning, 'The following physical objects have no shape. Please assert the shape for these objects or load the corresponding urdf files for the robot and environment, where the shapes are defined.'),
+  forall(member(O, POWithoutShape), print_message(warning, O))), !;
+   print_message(info, 'All physical objects in this episode have shape'), true.
+
+get_joints_without_proper_links(Joints) :-
+  findall(Joint,
+    ( has_type(Joint, urdf:'Joint'),
+    \+ has_parent_link(Joint, _),
+    \+ has_child_link(Joint, _)),
+    Joints),
+  length(Joints, Listlength),
+  ( Listlength > 0 -> print_message(warning, 'The following joints have no parent link or child link asserted. Please assert the corresponding links for the joints'),
+  forall(member(O, Joints), print_message(warning, O))), !;
+  print_message(info, 'All joints have their corresponding links'), true.
+
+load_logs(Folder) :-
+  get_path(Path, Folder),
+  remember(Path),
+  tf_mng_remember(Path).
+
```

Comment:
Add a predicate that calls all the test methods for the current episode (so that user can call it at the end of the neem generation)

Reply:
okay sure. I will update it and move the file

Reply:
Modified to satisfy the requested changes